### DISCUSSION TITLE
Build error

### DISCUSSION THREAD

**Discussion by kerkovits** [GrandOrgue/grandorgue#1434] [Developers]
@oleg68 These compiler messages on Fedora do not seem to be healthy:

```
/builddir/build/BUILD/grandorgue-3.10.1-1/src/core/GOHash.h:16:3: error: 'uint8_t' does not name a type
   16 |   uint8_t hash[20];
      |   ^~~~~~~
/builddir/build/BUILD/grandorgue-3.10.1-1/src/core/GOHash.h:14:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   13 | #include "contrib/sha1.h"
  +++ |+#include <cstdint>
   14 | 
/builddir/build/BUILD/grandorgue-3.10.1-1/src/core/archive/GOArchiveFile.cpp: In member function 'wxString GOArchiveFile::GetCurrentFileID() const':
/builddir/build/BUILD/grandorgue-3.10.1-1/src/core/archive/GOArchiveFile.cpp:87:3: error: 'uint64_t' was not declared in this scope
   87 |   uint64_t size = path_name.GetSize().GetValue();
      |   ^~~~~~~~
/builddir/build/BUILD/grandorgue-3.10.1-1/src/core/archive/GOArchiveFile.cpp:19:1: note: 'uint64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   18 | #include "config/GOConfigWriter.h"
  +++ |+#include <cstdint>
   19 | 
```
Do I need to set a flag to `cmake` or is there a bug in the source?

**Comment by oleg68** [comment]
I couldn't reproduce this problem on Fedora. These types are defined in `/usr/include/bits/stdint-uintn.h` that is included by default.

**Reply by kerkovits** [reply]
Thank you for your investigation. I could reproduce it again on COPR: https://copr.fedorainfracloud.org/coprs/kerkovits/GrandOrgue/builds/ However, according to your hints, I assume that the bug is not in GrandOrgue, but in the compiler, as it affects only the beta version. I hope that it will be resolved before Fedora 38 gets released.

**Comment by kerkovits** [comment]
@oleg68 I think, I found the reason: At https://gcc.gnu.org/gcc-13/porting_to.html it is stated, that:

> Some C++ Standard Library headers have been changed to **no longer include other headers that were being used internally** by the library. As such, C++ programs that used standard library components without including the right headers will no longer compile.
> 
> The following headers are used less widely in libstdc++ and may need to be included explicitly when compiling with GCC 13:
> 
> - &lt;string&gt; (for std::string, std::to_string, std::stoi etc.)
> - &lt;system_error&gt; (for std::error_code, std::error_category, std::system_error).
> - _**&lt;cstdint&gt; (for std::int8_t, std::int32_t etc.)**_
> - &lt;cstdio&gt; (for std::printf, std::fopen etc.)
> - &lt;cstdlib&gt; (for std::strtol, std::malloc etc.)

**Reply by oleg68** [reply]
Yes, this is the reason.

You are welcome to open PRs with  #include<cstdint>.

**Reply by kerkovits** [reply]
Just checking the source, there are many header files using `uintXX_t` types. A few of them `#include <stdint.h>` (e.g. /src/core/GORodgers.h), some of them include it implicitly by e.g. `#include "GOInt.h"` (like /src/core/GOGZipFormat.h), but some of them do not have it in the include chain (/src/grandorgue/sound/GOSoundCompress.h). It seems to be not trivial to get through the include chain to find, which headers need an additional `#include <stdint.h>`.
